### ISSUE TITLE
Understanding device winix discovery :  discovery.py

### ISSUE THREAD

**Issue by deepcoder** [edc1591/python-winix#3]
Hi,

Thank you for your work on this!

I am trying to understand how to discover Winix devices that are on local network.

Bit above my pay grade, your code in discovery.py.

Could you explain how it finds winix units or point me to a resource where I can learn more about this discovery code you have.

Thanks!

Dave

**Comment by steveb85** [comment]
Try this...
https://github.com/banzalik/homebridge-winix-c545 

I’m working on editing for the am90 which is a bit more complex, also out of my pay grade. Good luck!

**Comment by karepiu** [comment]
This code does not work (at least for me) the one that Steve85 pointed works. 

I believe it is Amazon Cognito which I believe is Amazon take on OAuth. 

It works on tokens. By user name and password you received few tokens that confirm your identity. Tokens have expire time and you should refresh the token using special refresh token once in a while. I will have code posted that does that in Python in next few days (or you can check Hfern version https://github.com/hfern/winix )

**Comment by deepcoder** [comment]
Hi,

Ideally, I would like to query the Winix devices directly. They have a web server running on port 8088. Anyone have any ideas with the login and password would be? I'm assuming it is baked in. Short of that, I would like to find the devices on the local network and check that they are alive. Evan's code seems to point to some way to discover the units. The cloud api worked fine, but it the unit goes off line, it is not clear when and if the cloud will invalidate the data. I check the timestamp in the returned cloud data and can see how old the data is, would like to know more about what and when the unit phone home. Thx.

**Comment by karepiu** [comment]
Yeah, I would like to have local control as well but only solutions I found online go to cloud. Hferns code seems to be reverse engineered which means their app uses cloud only as well.  I would not be surprised if  local api was used only for initial setup. If you find out how to control it locally let us know - I would really love to ditch cloud for simple, fast local control. 

You can try reverse engineer initial setup as well - most likely password/user name will be in plain text then.
